app/controllers/search_controller.rb

```diff
@@ -35,12 +45,21 @@ def category
         order_by sort_arr.first, sort_arr.last
       end
     end
-    if category && name
+
+    if category && name && params['featured']
+      old_category_repos = Repository.where(category: category, featured: true).distinct
+      new_category_repos = Category.where(name: name).distinct.includes(:repository).map(&:repository).select{ |repo| repo[:featured] == true }
+      @repositories = (old_category_repos + new_category_repos).uniq.sort!{|a,b|b.updated_at <=> a.updated_at}.paginate(:per_page=>12,:page=>params[:page]) do
```

Comment:
Don't need to `sort!` because 1) you aren't sorting in place, `sort` would work and 2) you can also do `sort_by(:updated_at)`

Reply:
cant use sort_by because `@repositories` is an array

Reply:
I think you mean you can't `order_by` on array's. [sort_by](https://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-sort_by) is part of Enumerable

Reply:
it's weird I know but i keep getting 
> wrong number of arguments (given 1, expected 0)

on this

 >@repositories = @repositories.select{|r| r.featured?}.uniq.sort_by(:updated_at)

when sort_by is used

Reply:
ohh i was trying sort_by() all the time and it's actually sort_by{} with curly brackets 😅